Support Bootstrap 5 floating labels. Fixes #573

thimo opened the pull request:
Add support for Bootstrap 5 floating labels. See https://github.com/bootstrap-ruby/bootstrap_form/issues/573.

<img width="1182" alt="CleanShot 2021-04-07 at 10 20 48@2x" src="https://user-images.githubusercontent.com/781488/113834273-f2387780-978a-11eb-9512-ab447988c962.png">

thimo commented:
Tests fail, but this should be fixed when merging https://github.com/bootstrap-ruby/bootstrap_form/pull/583.

lcreid commented on test/bootstrap_fields_test.rb:
```diff
@@ -428,4 +428,14 @@ class BootstrapFieldsTest < ActionView::TestCase
   end
+
+  test "can have a floating label" do
+    expected = <<-HTML.strip_heredoc
```
We can use `<<~` now, instead of `strip_heredoc`. We no longer support Ruby versions that _don't_ have `<<~`.

thimo replied:
Sounds great, I wasn't really aware of this (just copy/pasted an existing test). But this would be the first and only occurrence of `<<~` in the project where `strip_heredoc` is used 220 times. Mind if I change all of them in this PR? Or would you prefer to only have this one changed and create a new PR for the rest?

thimo replied:
Never mind, just saw you added the PR https://github.com/bootstrap-ruby/bootstrap_form/issues/585. Will pick that one up after this and change this single occurrence of `strip_heredoc`.

lcreid reviewed (commented):
I really appreciate all the work you've done. I'd just ask that you add something in the README (ideally with example) showing how to make floating labels with `bootstrap_form`. Thanks in advance.

thimo commented:
I've added a section "Floating Labels" to the "Form Styles", including a short example. Nothing too fancy. Please let me know if this can/should be improved on.

Additionally I've rebased my branch to the current `bootstrap-5` branch so tests would run, and also fixed one Rubocop offence.

lcreid commented on README.md:
````diff
@@ -693,6 +693,21 @@ The `custom` option can be used to replace the browser default styles for check
 <% end %>
 ```
 
+### Floating Labels
````
:fire: This is fine. Thanks!

lcreid reviewed (approved):
Nice extra test coverage. This looks great. Thanks again!
